mllib/src/main/scala/org/apache/spark/mllib/linalg/EigenValueDecomposition.scala

```diff
@@ -66,6 +66,9 @@ private[mllib] object EigenValueDecomposition {
     // ncv must be smaller than n
     val ncv = math.min(2 * k, n)
 
+    require(ncv * n.toLong < Integer.MAX_VALUE, "Product of 2*k*n must be smaller than " +
```

Comment:
Seems reasonable, though I have a few minor suggestions. I suppose `<=` is OK, technically? and if this size is worth checking, it's probably worth checking `ncv * (ncv + 8)` later. And then, maybe it's best to keep the check by the allocation so they don't get out of sync if someone changes them later.